nucliadb/tests/nucliadb/integration/test_vectorsets_api.py

```diff
@@ -127,3 +138,140 @@ async def test_learning_config_errors_are_proxied_correctly(
+    kbid = resp.json()["uuid"]
+
+    # Create a link resource
+    resp = await nucliadb_writer.post(
+        f"/kb/{kbid}/resources",
+        json={
+            "title": "link",
+            "description": "link",
+            "links": {
+                "link": {
+                    "uri": "https://en.wikipedia.org/wiki/Lionel_Messi",
+                }
+            },
+        },
+    )
+    assert resp.status_code == 201, resp.text
+    rid = resp.json()["uuid"]
+
+    # Ingest a processing broker message
+    bm = BrokerMessage(
+        kbid=kbid,
+        uuid=rid,
+        type=BrokerMessage.MessageType.AUTOCOMMIT,
+        source=BrokerMessage.MessageSource.PROCESSOR,
+    )
+    field = FieldID(field_type=FieldType.LINK, field="link")
+
+    text = "Lionel Messi is a football player."
+    et = ExtractedTextWrapper()
+    et.body.text = text
+    et.field.CopyFrom(field)
+    bm.extracted_text.append(et)
+
+    ev = ExtractedVectorsWrapper()
+    ev.field.CopyFrom(field)
+    ev.vectorset_id = "multilingual-2024-05-06"
+    vector = utils_pb2.Vector(
+        start=0,
+        end=len(text),
+        start_paragraph=0,
+        end_paragraph=len(text),
+    )
+    vector.vector.extend([1.0 for _ in range(1024)])
+    ev.vectors.vectors.vectors.append(vector)
+    bm.field_vectors.append(ev)
+
+    await inject_message(nucliadb_grpc, bm)
+
+    # Make a search and check that the document is found
+    await _check_semantic_search(nucliadb_reader, kbid)
+
+    # Now add a new vectorset
+    request = NewVectorSetRequest(
+        kbid=kbid,
+        vectorset_id="en-2024-05-06",
+        similarity=utils_pb2.VectorSimilarity.COSINE,
+        vector_dimension=1024,
+    )
+    resp: NewVectorSetResponse = await nucliadb_grpc.NewVectorSet(request)  # type: ignore
+    assert resp.status == NewVectorSetResponse.Status.OK  # type: ignore
+
+    # Ingest a new broker message as if it was coming from the migration
+    bm2 = BrokerMessage(
+        kbid=kbid,
+        uuid=rid,
+        type=BrokerMessage.MessageType.AUTOCOMMIT,
+        source=BrokerMessage.MessageSource.PROCESSOR,
+    )
+    field = FieldID(field_type=FieldType.LINK, field="link")
+
+    ev = ExtractedVectorsWrapper()
+    ev.field.CopyFrom(field)
+    ev.vectorset_id = "en-2024-05-06"
+    vector = utils_pb2.Vector(
+        start=0,
+        end=len(text),
+        start_paragraph=0,
+        end_paragraph=len(text),
+    )
+    vector.vector.extend([2.0 for _ in range(1024)])
+    ev.vectors.vectors.vectors.append(vector)
+    bm2.field_vectors.append(ev)
+
+    await inject_message(nucliadb_grpc, bm2)
+
+    # Make a search with the new vectorset and check that the document is found
+    await _check_semantic_search(nucliadb_reader, kbid, vectorset="en-2024-05-06")
+
+    # With the default vectorset the document should also be found
+    await _check_semantic_search(nucliadb_reader, kbid)
+
+
+async def _check_semantic_search(
```

Comment:
should we also check keyword search, just in case?
I wonder if the paragraphs/texts index is being deleted or not

Reply:
Sure, I'll fix the broker message and extend the test